src/Security/Authentication/Certificate/src/CertificateValidationCache.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace Microsoft.AspNetCore.Authentication.Certificate
+{
+    /// <summary>
+    /// MemoryCache based implementation used to store <see cref="AuthenticateResult"/> results after the certificate has been validated
+    /// </summary>
+    public class CertificateValidationCache : ICertificateValidationCache
+    {
+        private MemoryCache _cache;
+
+        public CertificateValidationCache()
+        {
+             _cache = new MemoryCache(new MemoryCacheOptions { SizeLimit = CacheSize });
+        }
+
+
+        /// <summary>
+        /// The expiration that should be used for entries in the MemoryCache, defaults to 30 minutes.
+        /// </summary>
+        public TimeSpan CacheEntryExpiration { get; set; } = TimeSpan.FromMinutes(30);
```

Comment:
```suggestion
        public TimeSpan CacheEntryExpiration { get; set; } = TimeSpan.FromMinutes(2);
```
2 minutes is the average idle connection timeout.